scala_util/src/main/scala/com/splicemachine/spark/splicemachine/SplicePartitioningUtils.scala

```diff
@@ -0,0 +1,495 @@
+package com.splicemachine.spark.splicemachine
+
+// note: this is marked as private in package org.apache.spark.sql.execution.datasources,
+// so we had to copy this out.
+// see
+// https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/PartitioningUtils.scala
+// this file is marked as excluded from spotbugs, see splice_protocol/findbugs-exclude.xml
+// to be able to detect wrong types for directory partitioning, we modified some part of the code, marked with
+// modified splicemachine { ... }
+
+/*
```

Comment:
The license goes typically at the very first line

Reply:
done